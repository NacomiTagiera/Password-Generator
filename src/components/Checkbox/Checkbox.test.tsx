import { render, screen } from '@testing-library/react';
import { userEvent } from '@testing-library/user-event';

import { CheckBox } from '.';

describe('CheckBox', () => {
  const label = 'Test Checkbox';

  it('renders with default props', () => {
    const { asFragment } = render(<CheckBox label={label} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('displays provided label and assigns aria-label', () => {
    render(<CheckBox label={label} />);

    screen.getByText(label);
    screen.getByLabelText(label);
  });

  it('injects native checkbox properties', () => {
    render(<CheckBox label={label} checked />);

    expect(screen.getByRole('checkbox')).toBeChecked();
  });

  it('invokes onChange function when clicked', async () => {
    const handleChange = jest.fn();
    render(<CheckBox label={label} onChange={handleChange} />);

    await userEvent.click(screen.getByRole('checkbox'));
    expect(handleChange).toHaveBeenCalledTimes(1);
  });
});
