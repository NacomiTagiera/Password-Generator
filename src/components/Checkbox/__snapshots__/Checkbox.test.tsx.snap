// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders with default props 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1llowvf-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        aria-label="Test Checkbox"
        class="PrivateSwitchBase-input css-1m9pwf3"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
    >
      Test Checkbox
    </span>
  </label>
</DocumentFragment>
`;
